Source/ThirdParty/ANGLE/src/common/log_utils.h

```diff
@@ -260,12 +260,17 @@ std::ostream &FmtHex(std::ostream &os, T value)
 #define ANGLE_EAT_STREAM_PARAMETERS \
     true ? static_cast<void>(0) : ::gl::priv::LogMessageVoidify() & (*::gl::priv::gSwallowStream)
 
+#if defined(ANGLE_ENABLE_ASSERTS)
+bool AreAssertionsEnabled(); // Implementation is in "common/debug.cpp"
+#endif  // defined(ANGLE_ENABLE_ASSERTS)
+
 // A macro asserting a condition and outputting failures to the debug log
 #if defined(ANGLE_ENABLE_ASSERTS)
 #    define ASSERT(expression)                                                                \
-        (expression ? static_cast<void>(0)                                                    \
-                    : (FATAL() << "\t! Assert failed in " << __FUNCTION__ << " (" << __FILE__ \
-                               << ":" << __LINE__ << "): " << #expression))
+        ((!(expression) && AreAssertionsEnabled())                                             \
+                    ? (FATAL() << "\t! Assert failed in " << __FUNCTION__ << " (" << __FILE__ \
```

Comment:
I think you inverted the condition. Now all assertions result in fatal crash if they pass.